common/components/forms/FormValidation.jsx

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import React from 'react';
+import _ from 'lodash'
+
+export type Validation = {|
+  checkFunc: (formState: any) => boolean,
+  errorMessage: string
+|};
+
+type Props = {|
+  validations: $ReadOnlyArray<Validation>,
```

Comment:
A general note that you can annotate the props as readonly by doing
```
type Props = {|
  +foo: number,
  +bar: string,
|};
```

https://flow.org/en/docs/types/interfaces/#toc-covariant-read-only-properties-on-interfaces

Reply:
Noted, thanks!